File: python/Parkinglot.py
#Saya Amelia Zalfa Julianti dengan NIM 2203999 mengerjakan Latihan 4 mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan kecurangan sesuai dengan apa yang telah dispesifikasikan. Aamiin.
from Car import Car
from Motorcycle import Motorcycle

class Parkinglot:
    __kapasitas = 0
    __jumlahKendaraan = []
    
    def __init__(self, kapasitas = 0, jumlahKendaraan = None):
        self.__kapasitas = kapasitas

    #Getter and Setter
    
    #Get kapasitas
    
    def get_kapasitas(self):
        return self.__kapasitas
    
    #set kapasitas
    def set_kapasitas(self, kapasitas):
        self.__kapasitas = kapasitas
    
    #Get jumlahKendaraan
    
    def get_jumlahKendaraan(self):
        return self.__jumlahKendaraan
    
    #set jumlahKendaraan
    def set_jumlahKendaraan(self, jumlahKendaraan):
        self.__jumlahKendaraan = jumlahKendaraan
    
    def add_jumlahKendaraan(self, jumlahKendaraan):
        self.__jumlahKendaraan.append(jumlahKendaraan)

File: cpp/Parkinglot.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan, termasuk kelas induk (kelas Sivitas Akademik) dari kelas Mahasiswa serta kelas yang dimiliki oleh mahasiswa (kelas Laptop dan Dosen)
#include <iostream>
#include <string>
#include <vector>
#include "Car.cpp"
#include "Motorcycle.cpp"


// Deklarasi namespace
using namespace std;

class ParkingLot
{
private:
    // Atribut yang digunakan pada kelas Mahasiswa terdiri dari nim dan fakultas yang merupakan identitas pembeda antar mahasiswa satu dengan lainnya. Setiap mahasiswa memiliki laptop dan dosen wali
    int kapasitas;
    vector<Car> jumlahKendaraanMobil;
	vector<Motorcycle>jumlahKendaraanMotor;
	string jumlahKeseluruhan;

public:
    // Constructor
    ParkingLot()
    {


    }


    // Constructor : Overloading
    ParkingLot(int kapasitas){  
        this->kapasitas = kapasitas;
    }


    // Constructor : Overloading
    ParkingLot(int kapasitas, vector<Car> jumlahKendaraanMobil){  
        this->kapasitas = kapasitas;
		this->jumlahKendaraanMobil = jumlahKendaraanMobil;
    }


    // Constructor : Overloading
   ParkingLot(int kapasitas, vector<Car> jumlahKendaraanMobil, vector<Motorcycle>jumlahKendaraanMotor){   
        this->kapasitas = kapasitas;
		this->jumlahKendaraanMobil = jumlahKendaraanMobil;
		this->jumlahKendaraanMotor = jumlahKendaraanMotor;
    }

	// Constructor : Overloading
   ParkingLot(int kapasitas, vector<Car> jumlahKendaraanMobil, vector<Motorcycle>jumlahKendaraanMotor, string jumlahKeseluruhan){   
        this->kapasitas = kapasitas;
		this->jumlahKendaraanMobil = jumlahKendaraanMobil;
		this->jumlahKendaraanMotor = jumlahKendaraanMotor;
		this->jumlahKeseluruhan = jumlahKeseluruhan;
    }

    // Method setter untuk set setiap nilai atribut pada kelas ParkingLot
    void setKapasitas(int kapasitas)
    {
        this->kapasitas = kapasitas;
    }


    void setJumlahKendaraanMobil(vector<Car> jumlahKendaraanMobil)
    {
        this->jumlahKendaraanMobil = jumlahKendaraanMobil;
    }


    void setJumlahKendaraanMotor(vector<Motorcycle>jumlahKendaraanMotor)
    {
        this->jumlahKendaraanMotor = jumlahKendaraanMotor;
    }

	void setJumlahKeseluruhan(string jumlahKeseluruhan)
    {
        this->jumlahKeseluruhan = jumlahKeseluruhan;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Mahasiswa
    int getKapasitas()
    {
        return kapasitas;
    }


    vector<Car> getJumlahKendaraanMobil()
    {
        return jumlahKendaraanMobil;
    }

	vector<Motorcycle> getJumlahKendaraanMotor()
    {
        return jumlahKendaraanMotor;
    }

	string getJumlahKeseluruhan()
    {
        return jumlahKeseluruhan;
    }



    // Destructor
    ~ParkingLot(){


    }
};
